astropy/time/core.py

```diff
@@ -464,6 +499,49 @@ def __repr__(self):
     def __str__(self):
         return str(getattr(self, self.format))
 
+    def strftime(self, format_spec):
+        """
+        Convert Time to a string according to a format specification.
+        See time.stftime documentation for format specification.
+
+        Parameters
+        ----------
+        format_spec : string
+            Format definition of return string.
+
+        Returns
+        -------
+        string
+            Formatted string according to the given format string.
+
+        """
+
+        if self.format == 'datetime':
```

Comment:
I don't think you want to handle `datetime` specially like this because you can (in theory) have a `Time` object with `format='datetime'` which is within a leap second.  Just because the format is `datetime` does not mean that the `value` has been computed.  This is because you can change the format of an existing Time object.